### ISSUE TITLE
Bug: declarations inside a lambda expression are incorrectly reported (AV1522)

### ISSUE THREAD

**Issue by bkoelman** [bkoelman/CSharpGuidelinesAnalyzer#89]
#### Analyzer
AV1522: AssignVariablesInSeparateStatementsAnalyzer

#### Repro steps
```csharp
Action act = () =>
{
    int iii = 5;
    int jjj = 4;

    Console.WriteLine(iii);
    Console.WriteLine(jjj);
};
```

#### Expected behavior
No report of AV1522.

#### Actual behavior
AV1522 is reported:
```
'act', 'iii' and 'jjj' are assigned in a single statement.
```
